packages/IngestAPI/src/main/kotlin/nz/govt/eop/Messages.kt

```diff
@@ -1,11 +1,12 @@
 package nz.govt.eop.messages
 
+import java.math.BigDecimal
 import java.time.Instant
 import nz.govt.eop.ingestapi.WaterAllocation
 
 data class WaterAllocationMessage(
     val areaId: String,
-    val amount: Int,
+    val amount: BigDecimal,
```

Comment:
This was based on Jooqs default mapping for Postgres numeric types, and matches what we use for `water_allocations.allocation_amount`

Reply:
👍 sweet, yeah avoid all the IEEE floating point pain